akka-docs/src/main/paradox/stream/operators/Source-or-Flow/concatLazy.md

```diff
@@ -0,0 +1,36 @@
+# concatLazy
+
+After completion of the original upstream the elements of the given source will be emitted.
+
+@ref[Fan-in operators](../index.md#fan-in-operators)
+
+## Signature
+
+@apidoc[Source.concat](Source) { scala="#concatLazy[U&gt;:Out,Mat2](that:akka.stream.Graph[akka.stream.SourceShape[U],Mat2]):FlowOps.this.Repr[U]" java="#concatLazy(akka.stream.Graph)" }
+@apidoc[Flow.concat](Flow) { scala="#concatLazy[U&gt;:Out,Mat2](that:akka.stream.Graph[akka.stream.SourceShape[U],Mat2]):FlowOps.this.Repr[U]" java="#concatLazy(akka.stream.Graph)" }
+
+
+## Description
+
+After completion of the original upstream the elements of the given source will be emitted.
+
+Both streams will be materialized together, but the concatenated source materialization can be deferred (and completely avoided if the stream fails or cancels before the first source has completed) by combining with @ref(Source.lazySource)[../Source/lazySource.md].
```

Comment:
```suggestion
Both streams will be materialized together, however, the given stream will be pulled for the first time only after the original upstream was completed. (In contrast, @ref(concat)[concat.md], introduces single-element buffers after both, original and given sources so that the given source is also pulled once immediately.)

To defer the materialization of the given source (or to completely avoid its materialization if the original upstream fails or cancels), wrap it into @ref(Source.lazySource)[../Source/lazySource.md].
```

Reply:
Maybe also point to `concatLazyMat`.

Reply:
I was trying to avoid talking about pull/push since the rest of the docs talk about it in a more abstract "apply backpressure"-way, but I agree that it makes things more clear here, I'll add that.

I looked and the operator docs doesn't mention the `mat` variants, so let's not do that here either to keep it consistent.

Reply:
Correction, the prepend docs do in fact refer to the mat version of prepend, so I take that back.